src/compiler/crystal/command/repl.cr

```diff
@@ -0,0 +1,48 @@
+# Implementation of the `crystal repl` command
+
+class Crystal::Command
+  private def repl
+    decompile = false
+    trace = false
+    stats = false
+
+    OptionParser.parse(options) do |opts|
+      opts.banner = "Usage: crystal repl [options] [files]\n\nOptions:"
+
+      opts.on("-d", "--decompile", "Show decompilation") do
+        decompile = true
+      end
+
+      opts.on("-s", "--status", "Show time taken to execute") do
+        stats = true
+      end
+
+      opts.on("-t", "--trace", "Trace execution") do
+        trace = true
+      end
```

Comment:
These are all experimental! Also `status` should be `stats` 😊 

I think the options there should match the options of the usual crystal compiled, being able to pass `--prelude`, `-D`, etc.